### PR TITLE
Deposit testing

### PR DESCRIPTION
**Pull request by cyberbono3**
|Resolves] #2955
This is outline for deposit end to end testing.

I run 2 validators sending 5 deposits each. 
---

### CODE REVIEW

File: contracts/deposit-contract/sendDepositTx/sendDeposits_test.go [hunk 1]
```diff
@@ -171,12 +175,12 @@ func TestEndtoEndDeposits(t *testing.T) {
 		if ok := trieutil.VerifyMerkleProof(
```
Comment (cyberbono3) [inline contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]: @nisdas this test raises the following [error](https://play.golang.org/p/dZVXWo1SIyB). Can you review to make sure  that correct parameters are used in VerifyMerkleProof.

Comment (nisdas) [reply contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]: this is wrong 
```
		depositTrie, err := trieutil.GenerateTrieFromItems([][]byte{leaf[:]}, int(params.BeaconConfig().DepositContractTreeDepth))
		if err != nil {
			t.Fatalf("Could not generate trie: %v", err)
		}
```
the trie needs all the deposits not just one

Comment (cyberbono3) [reply contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]: @nisdas thanks for your review. I have fixed that, but the VerifyMerkleBranch test still does not pass. This is the [error](https://play.golang.org/p/7R0Fm0m1U8X)

**Comment by cyberbono3** [comment]
@prestonvanloon   can you review it, please?

File: contracts/deposit-contract/sendDepositTx/sendDeposits_test.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	"github.com/prysmaticlabs/prysm/shared/trieutil"
+	"github.com/sirupsen/logrus"
+)
+
+func init() {
+	logrus.SetLevel(logrus.DebugLevel)
+	logrus.SetOutput(ioutil.Discard)
+}
+
+// sendDeposits sends deposits to deposit contract
```
Comment (rauljordan) [inline contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]:
```suggestion
```
no need for comment

File: contracts/deposit-contract/sendDepositTx/sendDeposits_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+		for j := uint64(0); j < numberOfDeposits; j++ {
+			tx, err := testAcc.Contract.Deposit(testAcc.TxOpts, data.PublicKey, data.WithdrawalCredentials, data.Signature, dataRoot)
+			if err != nil {
+				t.Fatalf("unable to send transaction to contract: %v", err)
+			}
+
+			testAcc.Backend.Commit()
+
+			//lgos do not show up in console
```
Comment (rauljordan) [inline contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]:
```suggestion
```

File: contracts/deposit-contract/sendDepositTx/sendDeposits_test.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+		if i == int(numberOfDeposits)-1 {
+			j++
+		}
+	}
+
+	encodedDeposits := make([][]byte, numberOfValidators*numberOfDeposits)
+	for i := 0; i < int(numberOfValidators); i++ {
+
+		hashedDeposit, err := ssz.HashTreeRoot(deposits[i].Data)
```
Comment (rauljordan) [inline contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]:
```suggestion
		hashedDeposit, err := ssz.SigningRoot(deposits[i].Data)
```

File: contracts/deposit-contract/sendDepositTx/sendDeposits_test.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+		}
+
+		if i == int(numberOfDeposits)-1 {
+			j++
+		}
+	}
+
+	encodedDeposits := make([][]byte, numberOfValidators*numberOfDeposits)
+	for i := 0; i < int(numberOfValidators); i++ {
+
```
Comment (rauljordan) [inline contracts/deposit-contract/sendDepositTx/sendDeposits_test.go]:
```suggestion
```

**Comment by cyberbono3** [comment]
@rauljordan  I applied requested changes. Please take a look

**Pull request merged**